### PR TITLE
doc: document etcdctl snapshot command

### PR DESCRIPTION
**Pull request by heyitsanthony**

### CODE REVIEW

File: etcdctl/READMEv3.md [hunk 1]
```diff
@@ -370,6 +370,64 @@ Simple reply
 [mirror]: ./doc/mirror_maker.md
 
 
+### SNAPSHOT [subcommand]
+
+SNAPSHOT takes a point-in-time snapshot of etcd and restores it into a fresh cluster.
```
Comment (xiang90) [inline etcdctl/READMEv3.md]: it seems like snapshot and restore happens atomically. Probably we shall rephrase? SNAPSHOT provides the functionality to take a point-in-time snapshot of a running etcd server and to restore from the saved snapshot into a fresh cluster.

File: etcdctl/READMEv3.md [hunk 2]
```diff
@@ -370,6 +370,64 @@ Simple reply
...
+
+### SNAPSHOT SAVE \<filename\>
+
+SNAPSHOT SAVE writes a point-in-time snapshot of the etcd backend database to a file.
+
+#### Return value
+
+- On success, the backend snapshot is written to the given file path.
+
+- Error string if mirroring failed. Exit code is non-zero.
```
Comment (xiang90) [inline etcdctl/READMEv3.md]: if snapshotting failed?

File: etcdctl/READMEv3.md [hunk 3]
```diff
@@ -370,6 +370,64 @@ Simple reply
...
+
+Save a snapshot to "snapshot.db":
+```
+./etcdctl snapshot save snapshot.db
+```
+
+
+### SNAPSHOT RESTORE [options] \<filename\>
+
+SNAPSHOT RESTORE creates an etcd data directory for an etcd cluster member from a backend snapshot and a new cluster configuration. Restoring the snapshot into each node for a new cluster configuration will initialize a new etcd cluster preloaded by the snapshot data.
```
Comment (xiang90) [inline etcdctl/READMEv3.md]: a backend database snapshot file? (we call this backend database snapshot at line 379)

Comment (xiang90) [inline etcdctl/READMEv3.md]: Restoring the snapshot into each member?

File: etcdctl/READMEv3.md [hunk 4]
```diff
@@ -370,6 +370,64 @@ Simple reply
...
+
+Save a snapshot and restore into a new 3 node cluster:
+```
+./etcdctl snapshot save snapshot.db
+
+/bin/etcdctl snapshot restore snapshot.db --initial-cluster-token etcd-cluster-1 --initial-advertise-peer-urls http://127.0.0.1:12380  --name sshot1 --initial-cluster 'sshot1=http://127.0.0.1:12380,sshot2=http://127.0.0.1:22380,sshot3=http://127.0.0.1:32380'
+/bin/etcdctl snapshot restore snapshot.db --initial-cluster-token etcd-cluster-1 --initial-advertise-peer-urls http://127.0.0.1:22380  --name sshot2 --initial-cluster 'sshot1=http://127.0.0.1:12380,sshot2=http://127.0.0.1:22380,sshot3=http://127.0.0.1:32380'
+/bin/etcdctl snapshot restore snapshot.db --initial-cluster-token etcd-cluster-1 --initial-advertise-peer-urls http://127.0.0.1:32380  --name sshot3 --initial-cluster 'sshot1=http://127.0.0.1:12380,sshot2=http://127.0.0.1:22380,sshot3=http://127.0.0.1:32380'
+```
+
```
Comment (xiang90) [inline etcdctl/READMEv3.md]: also add the etcd startup commands?

File: etcdctl/READMEv3.md [hunk 5]
```diff
@@ -370,6 +370,70 @@ Simple reply
...
+bin/etcdctl snapshot restore snapshot.db --initial-cluster-token etcd-cluster-1 --initial-advertise-peer-urls http://127.0.0.1:12380  --name sshot1 --initial-cluster 'sshot1=http://127.0.0.1:12380,sshot2=http://127.0.0.1:22380,sshot3=http://127.0.0.1:32380'
+bin/etcdctl snapshot restore snapshot.db --initial-cluster-token etcd-cluster-1 --initial-advertise-peer-urls http://127.0.0.1:22380  --name sshot2 --initial-cluster 'sshot1=http://127.0.0.1:12380,sshot2=http://127.0.0.1:22380,sshot3=http://127.0.0.1:32380'
+bin/etcdctl snapshot restore snapshot.db --initial-cluster-token etcd-cluster-1 --initial-advertise-peer-urls http://127.0.0.1:32380  --name sshot3 --initial-cluster 'sshot1=http://127.0.0.1:12380,sshot2=http://127.0.0.1:22380,sshot3=http://127.0.0.1:32380'
+
+# launch members
+bin/etcd --name sshot1 --listen-client-urls http://127.0.0.1:2379 --advertise-client-urls http://127.0.0.1:2379 --listen-peer-urls http://127.0.0.1:12380 --initial-advertise-peer-urls http://127.0.0.1:12380 --initial-cluster-token etcd-cluster-1 --initial-cluster 'sshot1=http://127.0.0.1:12380,sshot2=http://127.0.0.1:22380,sshot3=http://127.0.0.1:32380' --initial-cluster-state new &
```
Comment (xiang90) [inline etcdctl/READMEv3.md]: do we still need these initial flags?

**Comment by xiang90** [comment]
LGTM

**Comment by colhom** [comment]
Summary of in-person conversation early this afternoon:

Snapshot save / restore look good to me. I would like to see an `etcdctl snapshot ...` command which allows comparing multiple `snapshot save` outputs in order to determine which is the most recent. @xiang90 says that this is possible.

**Pull request merged**